### PR TITLE
pkg/trace: trace stats payload v2

### PR DESCRIPTION
**Pull request by raphaelgavache**
Add OtherTags field that include httpCode and origin:synthetics.
On the wire this is a map[string]Any, allowing different type as values

Next PR would contain:
- a breaking change with the stats endpoint URL changed
- stats endpoint on by default (removal of the feature flag)

### CODE REVIEW

File: pkg/trace/api/api_test.go [hunk 1]
```diff
@@ -537,12 +536,13 @@ func TestHandleStats(t *testing.T) {
...
-					Hits:     2,
-					Errors:   440,
-					Duration: 123,
+					Name:      "name",
+					Service:   "service",
+					Resource:  "/asd/r",
+					Hits:      2,
+					Errors:    440,
+					Duration:  123,
+					OtherTags: [][]string{[]string{"k1", "v1"}, []string{"k2", "v2"}},
```
Comment (richardstartin) [inline pkg/trace/api/api_test.go:545]: Can we just call this tags instead please?

**Comment by furmmon** [comment]
pb/stats.go and pb/stats_test.go are not generated and need to be reviewed carefully

File: pkg/trace/pb/stats_test.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	o = msgp.AppendBytes(o, z.OkSummary)
+	o = append(o, 0xac, 0x45, 0x72, 0x72, 0x6f, 0x72, 0x53, 0x75, 0x6d, 0x6d, 0x61, 0x72, 0x79)
+	o = msgp.AppendBytes(o, z.ErrorSummary)
+	return o
+}
+
+func TestMarshalUnmarshalTags(t *testing.T) {
+	bts := marshalNoTags(ClientGroupedStats{}, nil)
+	// string "Tags"
+	bts = append(bts, 0xa4, 0x54, 0x61, 0x67, 0x73)
```
Comment (furmmon) [inline pkg/trace/pb/stats_test.go:42]: test on all supported typess

File: pkg/trace/stats/stats.go [hunk 3]
```diff
@@ -65,11 +65,14 @@ type Distribution struct {
...
 // which is of the form: name|measure|aggr
 // for example: serve|duration|service:webserver
-func GrainKey(name, measure string, aggr Aggregation) string {
+func GrainKey(name, measure string, aggr Aggregation, otherTags [][]string) string {
 	b := strings.Builder{}
 	// +2 for "|" separators
 	size := len(name) + 1 + len(measure) + 1
 	size += aggr.KeyLen()
+	if len(otherTags) > 0 {
+		size += len(otherTags[0]) * len(otherTags)
```
Comment (piochelepiotr) [inline pkg/trace/stats/stats.go:74]: isn't the size dependent on the size of the strings also?

**Review by piochelepiotr — APPROVED** [review]

File: pkg/trace/filters/replacer.go [hunk 4]
```diff
@@ -60,9 +58,10 @@ func (f Replacer) ReplaceStatsGroup(b *pb.ClientGroupedStats) {
 			b.Resource = re.ReplaceAllString(b.Resource, str)
 			fallthrough
 		case "http.status_code":
-			strcode := re.ReplaceAllString(strconv.Itoa(int(b.HTTPStatusCode)), str)
-			if code, err := strconv.Atoi(strcode); err == nil {
-				b.HTTPStatusCode = uint32(code)
+			for _, t := range b.Tags {
+				if t.Name == "http.status_code" {
```
Comment (richardstartin) [inline pkg/trace/filters/replacer.go:62]: We were using "HTTPStatusCode" in the tracer - is this what you need the tag to be named?

**Pull request closed without merging**